data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/router/RouterCopyRecordStrategy.java

```diff
@@ -67,8 +93,24 @@ public Record getRecord(Record record) {
         } else if (record.getData() instanceof Event) {
             try {
                 final Event recordEvent = (Event) record.getData();
-                Event newRecordEvent = JacksonEvent.fromEvent(recordEvent);
-                return new Record<>(newRecordEvent);
+                JacksonEvent newRecordEvent;
+                Record newRecord;
+                DefaultEventHandle eventHandle = (DefaultEventHandle)recordEvent.getEventHandle();
+                if (eventHandle != null) {
+                    final EventMetadata eventMetadata = recordEvent.getMetadata();
+                    final EventBuilder eventBuilder = (EventBuilder) eventFactory.eventBuilder(EventBuilder.class).withEventMetadata(eventMetadata).withData(recordEvent.toMap());
+                    newRecordEvent = (JacksonEvent) eventBuilder.build();
+
+                    eventHandle.getAcknowledgementSet().add(newRecordEvent);
+                    newRecord = new Record<>(newRecordEvent);
+                    acquireEventReference(newRecord);
+                } else {
+                    // TODO we should have a way to create from factory
+                    // even when acknowledgements are not used
+                    newRecordEvent = JacksonEvent.fromEvent(recordEvent);
+                    newRecord = new Record<>(newRecordEvent);
```

Comment:
General question for myself, why do we not `acquireEventReference` in this code path?

Reply:
This is the "legacy" code path we plan to deprecate in future. And in this path, there will not be acknowledgements. When there are acknowledgements, event handle would be there and the other path would be taken.